src/codegen/doc/mod.rs

```diff
@@ -0,0 +1,255 @@
+use std::io::{Result, Write};
+
+use analysis::namespaces::MAIN;
+use env::Env;
+use file_saver::save_to_file;
+use library::*;
+use library::Type as LType;
+use writer::primitives;
+
+use stripper_interface as stripper;
+use stripper_interface::Type as SType;
+use stripper_interface::TypeStruct;
+use stripper_interface::{FILE, /*FILE_COMMENT, */MOD_COMMENT};
+
+pub trait ToStripperType {
+    fn convert(&self) -> stripper::TypeStruct;
+    fn doc(&self) -> Option<String>;
+    fn doc_deprecated(&self) -> Option<String>;
+}
+
+macro_rules! impl_to_stripper_type {
+    ($ty:ident, $enum_var:ident) => {
+        impl ToStripperType for $ty {
+            fn convert(&self) -> TypeStruct {
+                TypeStruct::new(SType::$enum_var, &self.name)
+            }
+
+            fn doc(&self) -> Option<String> {
+                self.doc.clone()
+            }
+
+            fn doc_deprecated(&self) -> Option<String> {
+                self.doc_deprecated.clone()
+            }
+        }
+    }
+}
+
+impl_to_stripper_type!(Alias, Struct);
+impl_to_stripper_type!(Constant, Const);
+impl_to_stripper_type!(Member, Variant);
+impl_to_stripper_type!(Enumeration, Enum);
+impl_to_stripper_type!(Bitfield, Type);
+impl_to_stripper_type!(Record, Type);
+impl_to_stripper_type!(Field, Variant);
+impl_to_stripper_type!(Union, Struct);
+impl_to_stripper_type!(Function, Fn);
+impl_to_stripper_type!(Interface, Trait);
+impl_to_stripper_type!(Class, Struct);
+impl_to_stripper_type!(Namespace, Mod);
+
+const DEPRECATED : &'static str = "/// ====>> DEPRECATED <<====";
```

Comment:
I don't know why the previous comment is gone, it will be more robust to add a `write_deprecated` function, which would emit
```
///
/// # Deprecated
///
```
This can't be done with just a const because of indents.